src/main/scala/com/scality/clueso/SparkUtils.scala

```diff
@@ -0,0 +1,129 @@
+      .config("spark.hadoop.fs.s3a.buffer.dir", "./tmp")
+      .config("spark.hadoop.fs.s3a.connection.ssl.enabled", config.s3SslEnabled)
+      .config("spark.hadoop.fs.s3a.endpoint", config.s3Endpoint)
+      .config("spark.hadoop.fs.s3a.impl", "org.apache.hadoop.fs.s3a.S3AFileSystem")
+      .config("spark.hadoop.fs.s3a.access.key", config.s3AccessKey)
+      .config("spark.hadoop.fs.s3a.secret.key", config.s3SecretKey)
+      .config("spark.hadoop.fs.s3a.path.style.access", config.s3PathStyleAccess)
+      .config("spark.sql.streaming.metricsEnabled", "true")
+      .config("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
+  }
+
+  def confSparkSession(spark : SparkSession, config : CluesoConfig) = {
+    spark.conf.set("spark.sql.parquet.cacheMetadata", "false")
+    spark.conf.set("spark.hadoop.fs.s3a.fast.upload", "true")
+    spark.conf.set("spark.hadoop.fs.s3a.buffer.dir", "/tmp")
+    spark.conf.set("spark.hadoop.fs.s3a.connection.ssl.enabled", config.s3SslEnabled)
+    spark.conf.set("spark.hadoop.fs.s3a.endpoint", config.s3Endpoint)
+    spark.conf.set("spark.hadoop.fs.s3a.impl", "org.apache.hadoop.fs.s3a.S3AFileSystem")
+    spark.conf.set("spark.hadoop.fs.s3a.access.key", config.s3AccessKey)
+    spark.conf.set("spark.hadoop.fs.s3a.secret.key", config.s3SecretKey)
+    spark.conf.set("spark.hadoop.fs.s3a.path.style.access", config.s3PathStyleAccess)
+    spark.conf.set("fs.s3a.path.style.access", config.s3PathStyleAccess)
+    spark.conf.set("spark.sql.streaming.metricsEnabled", "true")
+    spark.conf.set("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
+  }
+
+  import scala.util.parsing.json.JSONObject
+
+  def convertRowToJSON(row: Row): String = {
+    val m = row.getValuesMap(row.schema.fieldNames)
+    JSONObject(m).toString().replaceAll("`","")
+  }
+
+  def getQueryResults(spark : SparkSession, queryExecutor: MetadataQueryExecutor, query : MetadataQuery) = {
+    val result = queryExecutor.execute(query)
+
+    val resultArray = try {
+      val jsonResults = result.toJSON.take(query.limit)
+      if (jsonResults.size > 0) {
+        jsonResults.map(_.replaceAll("`",""))
+      } else {
+        Array[String]()
+      }
+    } catch {
+      case e:IOException => {
+        println(e)
+        Array[String]()
+      }
+    }
+
+    resultArray
+  }
+
+
+  val parquetFilesFilter = new PathFilter {
+    override def accept(path: Path): Boolean = path.getName.endsWith(".parquet")
+  }
+
+  def fillNonExistingColumns(myCols: Set[String], allCols: Set[String]) = {
+    allCols.toList.map(x => x match {
+      case x if myCols.contains(x) => col(x)
+      case _ => lit(null).as(x)
+    })
+  }
+
+  def getParquetFilesStats(fs: FileSystem, path : String) = {
+    val fsPath = new Path(path)
+    val statusList = if (!fs.exists(fsPath)) {
+      Array[FileStatus]()
+    } else {
+      fs.listStatus(new Path(path), parquetFilesFilter)
+    }
+
+    val fileCount = statusList.count(_ => true)
```

Comment:
why need a lambda function for count?

Reply:
because it accepts a filter function that allows to specify which entries get counted or not, as we want them all, had to put that 'always' true function